fhir-persistence-schema/docs/DB2MultiTenancy.md

```diff
@@ -0,0 +1,140 @@
+| TENANT_KEY_ID | Integer | 4 | No | The ID is from the sequence, and is used to identify the unique entry in the row |
+| MT_ID | Integer | 4 | No | Matches to the MT_ID in the TENANTS table |
+| TENANT_SALT | VARCHAR | 44 | No | A 32 byte random value used as a salt when computing the SHA-256 tenant hash |
+| TENANT_HASH | VARBINARY | 32 | No | Each tenant's data is encrypted. |
+
+The tenant_key is a 32 byte random value created when the tenant is provisioned. This is a secret, and must be passed by the client to the database in order to obtain access to a tenant. A tenant may have multiple tenant keys to support key rotation. An example of the tenant key is `GQUHRi0ETFw0Xe+p1VtVDki4JFZs885bGM35hXRyq/E=`
+
+Multiple keys per tenant are supported, and key rotation includes adding, and then subsequently removing prior keys. 
+
+**Table: VERSION_HISTORY**
+
+The VERSION_HISTORY table enables Tenant version isolation, so upgrades and patches are separately applied to each tenant's schema. 
+
+| Column Name | Data Type | Length | Null | Purpose |
+|----------|---------------|---------------|---------------|---------------------------------------------|
+| SCHEMA_NAME | VARCHAR | 64 | No | The name of the schema where the object resides|
+| OBJECT_TYPE | VARCHAR | 16 | No | The type of the object - Procedure, Sequence, Table, TableSpace, Variable |
+| OBJECT_NAME | VARCHAR | 64 | No | The name of the object in the schema |
+| VERSION | INTEGER | 4 | No | The version starting at one and incrementing with every version change |
+| APPLIED | TIMESTAMP | 10 | No | The time of the latest object update |
+
+### Schema: Tenant-Specific
+
+The Tenant specific data is partitioned and isolated in a customer specific tablespace.  The tenant data is not mixed with other tenant data physically using the tablespace isolation. 
+
+Tenants are never able to see each other's data, each user is only able to see a single tenant's data. 
+
+Each tenant's data is able to be moved, backedup, restored or exported a tenant-at-a-time.  This approach allows tenant independent restore points.
+
+The isolated data supports tenant independent query plan caches, scaling and disambiguating tenant specific disk and space usage. The performance metrics are tenant specific, and diagnostics are tenant specific. 
+
+**Table: Resources**
+
+Each resource is contains a root RESOURCE and LOGICAL RESOURCE, such as VISIONPRESCRIPTION:
+
+```
+VISIONPRESCRIPTION_RESOURCES
+VISIONPRESCRIPTION_LOGICAL_RESOURCES
+```
+
+These values are mapped to specific FHIR Search value types, supporting parameter searches: 
+
+```
+VISIONPRESCRIPTION_COMPOSITES
+VISIONPRESCRIPTION_DATE_VALUES
+VISIONPRESCRIPTION_LATLNG_VALUES
+VISIONPRESCRIPTION_NUMBER_VALUES
+VISIONPRESCRIPTION_QUANTITY_VALUES
+VISIONPRESCRIPTION_STR_VALUES
+VISIONPRESCRIPTION_TOKEN_VALUES
+```
+
+There are also global tables which enable a global search using the logical_resource and resource.  
```

Comment:
Not sure if I'm misreading this or not, but just to clarify:  there is a global `LOGICAL_RESOURCES` table, but no global `RESOURCES` table.

Reply:
yes - that's right - and the global is still per tenant.  I'll elaborate a bit more on 101. (actually a lot)...